internal/resolve/catalog_test.go

```diff
@@ -963,3 +963,150 @@ func TestMultipleChannels(t *testing.T) {
 	assert.Equal(t, bsemver.MustParse("2.0.0"), *gotVersion)
 	assert.Equal(t, ptr.To(packageDeprecation(pkgName)), gotDeprecation)
 }
+
+func TestAllCatalogsDisabled(t *testing.T) {
+	pkgName := randPkg()
+	listCatalogs := func(ctx context.Context, options ...client.ListOption) ([]catalogd.ClusterCatalog, error) {
+		return []catalogd.ClusterCatalog{
+			{
+				Spec: catalogd.ClusterCatalogSpec{
+					Availability: "Disabled",
+				},
+			},
+			{
+				Spec: catalogd.ClusterCatalogSpec{
+					Availability: "Disabled",
+				},
+			},
+		}, nil
+	}
+
+	getPackage := func(ctx context.Context, cat *catalogd.ClusterCatalog, packageName string) (*declcfg.DeclarativeConfig, error) {
+		return genPackage(pkgName), nil
+	}
```

Comment:
The expectation is that this function would never even be called. Perhaps we should panic?

```suggestion
	getPackage := func(ctx context.Context, cat *catalogd.ClusterCatalog, packageName string) (*declcfg.DeclarativeConfig, error) {
		panic("getPackage should never be called when all catalogs are disabled")
	}
```

Reply:
Implemented the panic in getPackage when called for a disabled catalog to ensure that disabled catalogs are not processed.